File: backend/src/application/commands/create_rule.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from src.domain.entities.rule import Action, Condition
from src.domain.value_objects.rule_type import RuleType


@dataclass(slots=True)
class CreateRuleCommand:
    """创建规则命令"""
    project_id: str
    owner_id: str
    name: str
    description: str
    rule_type: RuleType
    conditions: list[Condition] = field(default_factory=list)
    actions: list[Action] = field(default_factory=list)
    priority: int = 0
    is_active: bool = True


@dataclass(slots=True)
class UpdateRuleCommand:
    """更新规则命令"""
    rule_id: str
    owner_id: str
    name: str | None = None
    description: str | None = None
    conditions: list[Condition] | None = None
    actions: list[Action] | None = None
    priority: int | None = None
    is_active: bool | None = None


@dataclass(slots=True)
class DeleteRuleCommand:
    """删除规则命令"""
    rule_id: str
    owner_id: str


@dataclass(slots=True)
class ActivateRuleCommand:
    """激活规则命令"""
    rule_id: str
    owner_id: str


@dataclass(slots=True)
class DeactivateRuleCommand:
    """停用规则命令"""
    rule_id: str
    owner_id: str

File: backend/src/application/commands/create_rule.ts
import { ProjectRepository } from '@/domain/ports/repositories/project_repository';
import { Rule } from '@/domain/services/reasoning/rule_engine';

export interface CreateRuleCommand {
  projectId: string;
  name: string;
  description: string;
  condition: string;
  action: string;
  priority: number;
  enabled: boolean;
}

export class CreateRuleHandler {
  constructor(private readonly projectRepository: ProjectRepository) {}

  async execute(command: CreateRuleCommand): Promise<Rule> {
    // Verify project exists
    const project = await this.projectRepository.getProject(command.projectId);
    if (!project) {
      throw new Error('Project not found');
    }

    // Generate rule ID
    const ruleId = `rule-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;

    // Create rule
    const rule: Rule = {
      id: ruleId,
      name: command.name,
      description: command.description,
      condition: command.condition,
      action: command.action,
      priority: command.priority,
      enabled: command.enabled,
    };

    // TODO: Persist rule to database
    // For now, we'll just return the rule

    return rule;
  }
}
